### PR TITLE
[12.0][IMP][FIX] web_widget_one2many_product_picker: Context and Subtotal

### PR DESCRIPTION
**Pull request by Tardo**
New context that contains search involved fields
Fixed a bug that don't uses discounts to calculate the subtotal price when 'show_discount' is enabled.
Fixed Readme.
Fixed a bug when save/discard the main record while the widget search process is not over yet.

cc @tecnativa TT27225

### CODE REVIEW

File: web_widget_one2many_product_picker/readme/USAGE.rst [hunk 1]
```diff
@@ -45,7 +45,7 @@ Other example for 'purchase.order.line' fields:
         nolabel="1"
         widget="one2many_product_picker"
         mode="form"
-        options="{'search': [{'name': _('Name'), 'domain': [['name', 'ilike', '$search']]}, {'name': _('Price'), 'domain': [['list_price', '=', $number_search]]}], 'field_map': {'name': 'name', 'product': 'product_id', 'product_uom': 'product_uom', 'price': 'price_unit', 'parent_id': 'order_id', 'product_uom_qty': 'product_qty'}, 'groups': [{'name': _('Desk'), 'domain': [['name', 'ilike', 'desk']], 'order': {'name': 'id', 'asc': true}}, {'name': _('Chairs'), 'domain': [['name', 'ilike', 'chair']]}]}"
+        options="{'search': [{'name': _('Name'), 'domain': [['name', 'ilike', '$search']]}, {'name': _('Price'), 'domain': [['list_price', '=', $number_search]]}], 'field_map': {'product_uom_qty': 'product_qty'}, 'groups': [{'name': _('Desk'), 'domain': [['name', 'ilike', 'desk']], 'order': {'name': 'id', 'asc': true}}, {'name': _('Chairs'), 'domain': [['name', 'ilike', 'chair']]}]}"
```
Comment (pedrobaeza) [inline web_widget_one2many_product_picker/readme/USAGE.rst]: Why is this still in USAGE.rst and not in CONFIGURE.rst?

**Comment by Tardo** [comment]
Changes done. I hope that now all content are in their correct place :/

**Review by pedrobaeza — APPROVED** [review]

**Comment by Tardo** [comment]
This is ready for final reviews

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Pull request merged**